openedx/features/partners/give2asia/tests/test_helpers.py

```diff
@@ -0,0 +1,201 @@
+
+    def test_get_partner_recommended_courses_with_invalid_enrollment_end_date(self):
+        """
+        Test empty list of recommendation is returned if enrolment end is in the past
+        :return : list of recommended courses
+        """
+        # get a future date
+        enrollment_date = datetime.now() + timedelta(days=-2)
+        PartnerCourseOverviewFactory(id=self.course.id, enrollment_start=enrollment_date)
+
+        recommended_courses = helpers.get_partner_recommended_courses(self.PARTNER_SLUG, self.user)
+        self.assertEqual(len(recommended_courses), 0)
+
+    @patch('openedx.features.partners.helpers.log.info')
+    def test_auto_join_partner_community_with_valid_partner_community(self, mock_log_info):
+        """
+        log.info method gets called when task is successfully
+        added to celery. Using mock make sure that log.info gets
+        called
+        Create a partner and partner community
+         :return : None
+        """
+        PartnerCommunityFactory(partner=self.partner)
+        helpers.auto_join_partner_community(self.partner, self.user)
+        assert mock_log_info.called_once
+
+    @patch('openedx.features.partners.helpers.log.info')
+    def test_auto_join_partner_community_with_multiple_partner_community(self, mock_log_info):
+        """
+        Test that log.info method gets called for all added
+        communities for a partner
+        Create 2 partner communities
+        :return : None
+        """
+        PartnerCommunityFactory(partner=self.partner)
+        PartnerCommunityFactory(partner=self.partner)
+        helpers.auto_join_partner_community(self.partner, self.user)
+        self.assertEqual(mock_log_info.call_count, 2)
+
+    @patch('openedx.features.partners.helpers.log.info')
+    def test_auto_join_partner_community_without_partner_community(self, mock_log_info):
+        """
+        log.info method gets called when task is successfully
+        added to celery. Using mock make sure that log.info gets
+        called
+        :return : None
+        """
+        helpers.auto_join_partner_community(self.partner, self.user)
+        assert mock_log_info.not_called
+
+    def test_get_partner_from_user_with_partner(self):
+        """
+        Test that valid partner object is returned against user
+        :return: partner
+        """
+        self.assertEqual(helpers.get_partner_from_user(self.user), self.partner)
+
+    def test_get_partner_from_user_without_partner_user_relation(self):
+        """
+        Test that none is returned if a bridget between user and partner
+        is not created
+        @return: partner
+        """
+        user = UserFactory()
+        self.assertEqual(helpers.get_partner_from_user(user), None)
+
+    def test_user_has_performance_access_with_permission(self):
+        """
+        Test that a valid user has access to
+        Give user permission to access performance page
```

Comment:
Lets update doc string, so that it is meaningful

Reply:
done